File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque


n , m , start = map(int,input().split())
graph = [[] for i in range(n+1)]
visited_d = [0] * (n+1)
visited_b = [0] * (n+1)

for i in range(m):
  a ,b = map(int,input().split())
  graph[a].append(b)
  graph[b].append(a)

for i in range(1,n+1):
  graph[i].sort()

def dfs(start):
  visited_d[start] = True
  print(start , end=' ')

  for i in graph[start]:
    if not visited_d[i]:
      dfs(i)

def bfs(start):
  q = deque([start])
  visited_b[start] = True

  while q:
    now = q.popleft()  
    print(now, end=' ')

    for i in graph[now]:
      if not visited_b[i]:
        q.append(i)
        visited_b[i] = True

dfs(start)
print()
bfs(start)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    static int N;
    static int min = Integer.MAX_VALUE;

    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int v = Integer.parseInt(st.nextToken());

        int[][] graph = new int[n+1][n+1];

        for (int i=0; i<m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph[a][b] = graph[b][a] = 1;
        }

        boolean[] visited = new boolean[n+1];
        dfs(graph, v, visited);
        System.out.println();
        visited = new boolean[n + 1];
        bfs(graph, v, visited);
    }

    public static void bfs(int[][] graph, int start, boolean[] visited) {
        Queue<Integer> q = new LinkedList<>();
        q.add(start);
        visited[start] = true;

        while(!q.isEmpty()) {
            int now = q.poll();
            System.out.print(now + " ");

            for(int next=1; next<graph[now].length; next++) {
                int value = graph[now][next];
                if(value == 1 && !visited[next]) {
                    q.add(next);
                    visited[next] = true;
                }
            }
        }


    }

    public static void dfs(int[][] graph, int start, boolean[] visited) {
        visited[start] = true;
        System.out.print(start + " ");

        for (int next=1; next<graph[start].length; next++) {
            int value = graph[start][next];
            if (value == 1 && !visited[next]) {
                dfs(graph, next, visited);
            }
        }
    }

}